packages/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,738 @@
+import { StackPanelClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { classNames, Direction, RtlScrollConverter } from "@microsoft/fast-web-utilities";
+import { canUseDOM } from "exenv-es6";
+import { get, isEqual, isNil, memoize, merge } from "lodash-es";
+import React from "react";
+import ReactDOM from "react-dom";
+import { toPx } from "@microsoft/fast-jss-utilities";
+import {
+    DisplayNamePrefix,
+    IntersectionObserverEntry,
+    ResizeObserverClassDefinition,
+    ResizeObserverEntry,
+} from "../utilities";
+import {
+    StackPanelHandledProps,
+    stackPanelOrientation,
+    StackPanelProps,
+    StackPanelUnhandledProps,
+} from "./stack-panel.props";
+import { isFunction } from "util";
+
+export interface StackPanelState {
+    disabled: boolean;
+    isScrolling: boolean;
+    renderedRangeStartIndex: number;
+    renderedRangeEndIndex: number;
+    direction: Direction;
+}
+
+interface ItemPosition {
```

Comment:
What do the values in `ItemPosition` represent? Are these pixel offset values? Is some of this information derived?

Reply:
Added comment "Used to store the pixel coordinates and span of items along the stackpanel's axis".  Basically it's used to create a lookup table of the coordinates and spans of the children on mount and update (see "updateItemPositions()") so we don't have to recalculate on scroll events.